java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java

```diff
@@ -45,21 +39,19 @@
     List<Pair<Pattern, List<JsonPath>>> compiledPseudonymizationsWithOriginals;
     List<Pattern> compiledAllowedEndpoints;
 
-    @Getter(onMethod_ = {@VisibleForTesting})
-    Configuration jsonConfiguration;
-
-    //TODO: inject
-    HashUtils hashUtils = new HashUtils();
-
-    public void initConfiguration() {
-        //jackson here because it's our common JSON stack, but adds dependency beyond the one pkg'd
-        // with JsonPath.
-        this.jsonConfiguration = Configuration.defaultConfiguration()
-            .jsonProvider(new JacksonJsonProvider())
-            .mappingProvider(new JacksonMappingProvider())
-            .setOptions(Option.SUPPRESS_EXCEPTIONS); //we specifically want to ignore PATH_NOT_FOUND cases
+    @AssistedInject
+    public SanitizerImpl(HashUtils hashUtils, @Assisted Options options) {
+        this.hashUtils = hashUtils;
+        this.options = options;
     }
 
+    @Getter(onMethod_ = {@VisibleForTesting})
+    @Inject Configuration jsonConfiguration;
```

Comment:
hmm, maybe this should be part of the `@AssistedInject` constructor.
`public SanitizerImpl(HashUtils hashUtils, Configuration jsonConfiguration, @Assisted Options options) {`
Not sure if it's going to work when creating through the factory, I bet jsonConfiguration will be null.